sql/planbuilder/builder.go

```diff
@@ -35,10 +41,43 @@ type Builder struct {
 	viewCtx         *ViewContext
 	procCtx         *ProcContext
 	triggerCtx      *TriggerContext
+	bindCtx         *BindvarContext
 	insertActive    bool
 	nesting         int
 }
 
+// ViewContext overwrites database root source of nested
+// calls.
+type BindvarContext struct {
+	Bindings map[string]*querypb.BindVariable
+	used     map[string]struct{}
+	// skip indicates that we are resolving plan names,
+	// but will not error for missing bindvar replacement
+	skip bool
+}
+
+func (bv *BindvarContext) GetSubstitute(s string) (*querypb.BindVariable, bool) {
+	if bv.Bindings != nil {
+		ret, ok := bv.Bindings[s]
+		bv.used[s] = struct{}{}
```

Comment:
You're marking it used without knowing if it was there, problem?

Reply:
We have some tests that define a binding, reference it multiple times, and expect not to error. I am not sure if that's a normal thing for SQL clients, but just made the code comply with our existing tests.